framework/Runners/ExternalRunner.py

```diff
@@ -1,274 +0,0 @@
-
-    if os.path.exists(executable):
-      executableFile = open(executable, "r")
-
-      firstTwoChars = executableFile.read(2)
-
-      if firstTwoChars == "#!":
-        realExecutable = shlex.split(executableFile.readline())
-        self.raiseAMessage("reading #! to find executable:" + repr(realExecutable))
-        # The below code should work, and would be better than findMsys,
-        # but it doesn't work.
-        # winExecutable = subprocess.check_output(['cygpath','-w',realExecutable[0]],shell=True).rstrip()
-        # print("winExecutable",winExecutable)
-        # realExecutable[0] = winExecutable
-        def findMsys():
-          """
-            Function to try and figure out where the MSYS64 is.
-            @ In, None
-            @ Out, dir, String, If not None, the directory where msys is.
-          """
-          dir = os.getcwd()
-          head, tail = os.path.split(dir)
-          while True:
-            if tail.lower().startswith("msys"):
-              return dir
-            dir = head
-            head, tail = os.path.split(dir)
-          return None
-        msysDir = findMsys()
-        if msysDir is not None:
-          beginExecutable = realExecutable[0]
-          if beginExecutable.startswith("/"):
-            beginExecutable = beginExecutable.lstrip("/")
-          winExecutable = os.path.join(msysDir, beginExecutable)
-          self.raiseAMessage("winExecutable" + winExecutable)
-          realExecutable[0] = winExecutable
-        else:
-          self.raiseAWarning("Could not find msys in "+os.getcwd())
-        commandSplit = realExecutable + [executable] + commandSplit[1:]
-        return commandSplit
-    return origCommand
-
-
-  def start(self):
-    """
-      Function to run the driven code
-      @ In, None
-      @ Out, None
-    """
-    oldDir = os.getcwd()
-    os.chdir(self.workingDir)
-    localenv = dict(os.environ)
-    outFile = open(self.output,'w', self.bufferSize)
-    if platform.system() == 'Windows':
-      self.command = self.expandForWindows(self.command)
-      self.raiseAMessage("modified command to" + repr(self.command))
-    self.__process = utils.pickleSafeSubprocessPopen(self.command,shell=True,stdout=outFile,stderr=outFile,cwd=self.workingDir,env=localenv)
-    os.chdir(oldDir)
-    self.started = True
-
-  def kill(self):
-    """
-      Function to kill the subprocess of the driven code
-      @ In, None
-      @ Out, None
-    """
-    self.raiseAWarning("Terminating "+self.__process.pid+' '+self.command)
-    self.__process.terminate()
-
-  def getWorkingDir(self):
-    """
-      Function to get the working directory path
-      @ In, None
-      @ Out, workingDir, string, working directory
-    """
-    return self.workingDir
-
-  def getOutputFilename(self):
-    """
-      Function to get the output filenames
-      @ In, None
-      @ Out, self.output, string, output filename root
-    """
-    return os.path.join(self.workingDir,self.output)
```

Comment:
I thought this might come up and I had this debate internally as well. My reasoning was: does it make sense to have a piece of unused code sitting in the repository that will slowly become out of sync with updates made to the rest of the code in the event that we will want it back in the future?  My answer was no, the code still exists in the revision history and a developer will be to extract it by digging it out, but they will be aware that the code they are dealing with worked on a __previous__ version of RAVEN.

By virtue of leaving it in the repository, we need to maintain it as we do everything else which means writing hypothetical tests to test a hypothetically useful piece of code. The majority of the inner workings of this code have not been thrown away, but instead moved.

Sorry, this was a bit of a digression, I contemplated deleting this post, but I still think it is useful information for others to see.